contacts.py

```diff
@@ -324,6 +367,7 @@ def _map_over_matching_contacts(self, prefix, ids, since, now, start_pos=0):
             for contact_id in filter(lambda x: x.startswith(prefix), ids.keys()):
                 for (floating_time, serial_number) in ids[contact_id]:
                     if _good_date(floating_time, since, now):
+                        # TODO-DAN is it ever the case that serial_number is None ?
```

Comment:
yes.  This was for backward compatibility, since we don't have any live data in the field we can discuss getting rid of it.

Reply:
To discuss getting rid of - I think we might as well, since there are other places I know we are assuming serial_number is present